src/intrusive.rs

```diff
@@ -36,67 +47,141 @@ struct Inner<T> {
 impl<T> List<T> {
     /// Create a new, empty event listener list.
     pub(super) fn new() -> Self {
-        Self(Mutex::new(Inner {
+        let inner = Inner {
             head: None,
             tail: None,
             next: None,
             len: 0,
             notified: 0,
-        }))
+        };
+
+        #[cfg(feature = "critical-section")]
+        {
```

Comment:
unnecessary parentheses

Reply:
You'd think, but when I remove these braces it throws a compile error.